### PR TITLE
Removed the custom splash screen

### LINKED ISSUE #254: Remove splash screen from ODK Collect
The splash screen was added to enable branding. There continue to be forks of Collect because the splash screen with a custom logo is not adequate. I would like to remove the splash screen because it is not sufficient and focus instead on the solutions in #233

### Notes

We should make it so the new [Android 12 Splash Screen](https://developer.android.com/guide/topics/ui/splash-screen) is shown instead of our one (investigation into this in #4592). As part of this work we should make app initialization run while the splash screen is up (using the [new API](https://developer.android.com/guide/topics/ui/splash-screen#suspend-drawing)).

This would be a perfect time to consider updating the Collect icon...

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #254

#### What has been done to verify that this works as intended?
I've played with the app to make sure it works without any crashes or bugs.

#### Why is this the best possible solution? Were any other approaches considered?
I removed the old custom splash screen and enabled the new one according to https://developer.android.com/guide/topics/ui/splash-screen. It should look like:

| Light mode  | Dark mode |
| ------------- | ------------- |
| Screenshot_1650922010 | Screenshot_1650922036 |

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Please verify that everything works well after removing the custom splash screen. It's important to use different android versions and maybe different screen sizes.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/res/drawable/splash_screen_icon.xml [hunk 1]
```diff
@@ -0,0 +1,314 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (grzesiek2010) [inline collect_app/src/main/res/drawable/splash_screen_icon.xml:1]: @lognaturel I had to add a vector version of the icon to display it in the splash screen. I think it looks pretty good (similar to the original one) but let me know if you have a better one.

File: buildSrc/src/main/java/dependencies/Dependencies.kt [hunk 2]
```diff
@@ -75,6 +75,7 @@ object Dependencies {
     const val slf4j_timber = "com.arcao:slf4j-timber:3.1@aar"
     const val emoji_java = "com.vdurmont:emoji-java:5.1.1"
     const val json_schema_validator = "com.networknt:json-schema-validator:1.0.66"
+    const val splashscreen = "androidx.core:core-splashscreen:1.0.0-beta02"
```
Comment (lognaturel) [inline buildSrc/src/main/java/dependencies/Dependencies.kt:78]: None of us love bringing in beta dependencies but I agree it's the way to go here.

**Review by lognaturel — APPROVED** [review]

File: collect_app/src/main/res/drawable/splash_screen_icon.xml (hunk 1 above)
Comment (lognaturel) [reply collect_app/src/main/res/drawable/splash_screen_icon.xml:1]: Works for me. I don't have a different one.

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1 and 8.1

Verified cases:
- app update when splash screen selected on the previous version (default and user's picture) 
- light/dark mode selected in device's settings
- screen rotation on new splash screen 
- minimize app when splash screen opened 
- import settings from old version when old splash screen is selected. 
- splash screen removed from user interface and access control/user settings

Collect icon is not visible on Android 5.1, only an empty page displays - I am not sure if we will want to fix it. Should I open a new issue? @grzesiek2010 @lognaturel

**Comment by srujner** [comment]
**Tested with success!**

Verified also on Androids 10 and 11

**Comment by grzesiek2010** [comment]
> Collect icon is not visible on Android 5.1, only an empty page displays - I am not sure if we will want to fix it. Should I open a new issue?

I think we can ignore it:
> On API 21-22, The icon isn't displayed until the application starts, only the background is visible.

https://developer.android.com/reference/androidx/core/splashscreen/SplashScreen#themes

**Pull request merged**